subxt/src/blocks/extrinsic_types.rs

```diff
@@ -558,6 +671,101 @@ impl<T: Config> ExtrinsicEvents<T> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct ExtrinsicSignedExtensions<'a, T: Config> {
+    signed_extensions: Vec<ExtrinsicSignedExtension<'a, T>>,
+    bytes: &'a [u8],
+    ty_id: u32,
+    metadata: &'a Metadata,
+    phantom: PhantomData<T>,
+}
+
+#[derive(Debug, Clone)]
+pub struct ExtrinsicSignedExtension<'a, T: Config> {
+    bytes: &'a [u8],
+    ty_id: u32,
+    name: &'a str,
+    metadata: &'a Metadata,
+    phantom: PhantomData<T>,
+}
+
+impl<'a, T: Config> ExtrinsicSignedExtensions<'a, T> {
```

Comment:
I'd be tempted to remove`decoded`, `bytes` and `signed_extensions` for now, and tweak `find`. The goal I think is to try to be consistent with `Extrinsics` here.

So I reckon the interface here should be something like:

```rust
// iterate and return each ExtrinsicSignedExtension (one can
// find by name or whatever using this interface):
iter()

// find and decode a specific signed extension just like we do for Extrinsics (leave this 
// for a separate PR since it's tricky and needs more work/thought):
find<S: SignedExtension>() -> Result<Option<S>, Error>

// The nice convenience helpers, which internally just self.iter().find() or whatever:
tip()
nonce()
```